arcgis-ios-sdk-samples/Utility Network/Display utility association/DisplayUtilityAssociationViewController.swift

```diff
@@ -0,0 +1,142 @@
+    private let connectivitySymbol = AGSSimpleLineSymbol(style: .dot, color: .red, width: 5)
+    
+    func loadUtilityNetwork() {
+        utilityNetwork.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get all the edges and junctions in the network.
+                let sourcesByType = Dictionary(grouping: self.utilityNetwork.definition.networkSources) { $0.sourceType }
+                let operationalLayers = self.mapView.map?.operationalLayers
+                // Add all edges that are not subnet lines to the map.
+                let edgeLayers = sourcesByType[.edge]!
+                    .filter { $0.sourceUsageType != .subnetLine }
+                    .map { AGSFeatureLayer(featureTable: $0.featureTable) }
+                operationalLayers?.addObjects(from: edgeLayers)
+                // Add all the junctions to the map.
+                let junctionLayers = sourcesByType[.junction]!.map { AGSFeatureLayer(featureTable: $0.featureTable) }
+                operationalLayers?.addObjects(from: junctionLayers)
+                
+                // Create a renderer for the associations.
+                let attachmentValue = AGSUniqueValue(description: "Attachment", label: "", symbol: self.attachmentSymbol, values: [AGSUtilityAssociationType.attachment])
+                let connectivityValue = AGSUniqueValue(description: "Connectivity", label: "", symbol: self.connectivitySymbol, values: [AGSUtilityAssociationType.connectivity])
+                self.associationsOverlay.renderer = AGSUniqueValueRenderer(fieldNames: ["AssociationType"], uniqueValues: [attachmentValue, connectivityValue], defaultLabel: "", defaultSymbol: nil)
+                
+                // Populate the legened.
+                self.attachmentSymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    guard let self = self else { return }
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self.attachmentBBI.image = image?.withRenderingMode(.alwaysOriginal)
+                        self.toolbar.items?.insert(self.attachmentBBI, at: 0)
+                    }
+                }
+                self.connectivitySymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    guard let self = self else { return }
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self.connectivityBBI.image = image?.withRenderingMode(.alwaysOriginal)
+                        self.toolbar.items?.insert(self.connectivityBBI, at: 4)
```

Comment:
This assumes the first swatch has already been created and the associated bar button item has been added to the toolbar, correct? Even though it may look like that happens every time, there's no guarantee of that, as far as I know. It would be better to use a dispatch group for these two operations and then create the items and assign them to the toolbar after both images have been created.

Reply:
@philium Updated! Please re-review, thanks!